bash_conditionals_loops/bash_conditionals_loops.md

```diff
@@ -0,0 +1,620 @@
+One of Bash's worst properties is that, unlike most programming languages which think of "true" statements as `1` and "false" statements as `0`, in Bash a "true" statement is assigned the numerical value `0` (except when evaluating mathematical expressions inside of double parentheses).
+
+</div>
+
+### If / Then Structure
+
+The basic structure of an if / then statement in Bash is:
+
+```
+if STATEMENT
+then RUN THIS COMMAND
+fi
+```
+The three lines together make up the conditional statement.
+
+* `if` tells the computer to check whether the statement following it is true or false
+* `then` tells the computer what to do if the statement is true
+* `fi` tells the computer that the conditional statement is completed.
+
+Let's see a working example:
+
+```
+if [ 5 -eq 05 ]
+then echo "integer equality"
+fi
+```
+
+<div class = "important">
+
+**Bash's `test` utility**
+
+A Bash test statement looks like this:
+
+`[ statement here ]`
+
+Notice that the statement is surrounded by square brackets with a space after the opening `[` and before the closing `]`
+
+</div>
+
+The statement `[ 5 -eq 05 ]` is checking if the numbers `5` and `05` are equal to each other. The command `echo "integer equality"` only runs if the numbers are equal. If you change the `05` to `04`, this code will not return any output.
+
+**`else`**
+
+You can also give instructions for what to do if the statement evaluates as false by adding a line starting with `else` after your `then` statement:
+
+```
+if [ 5 -eq 05 ]
+then echo "integer equality"
+else echo "these numbers are not equal"
+fi
+```
+
+Running this code exactly as is will give you the same output you got before. But now if you change the `5` or `05` so that the integers are not equal, you will get a message telling you that!
+
+**`elif`**
+
+The command `elif` is a shortening of the words "else if". This allows you to tell the computer to check another statement if the first statement is false. The `elif` line is followed by its own instance of `then` with instructions for what to do if that statement is true:
+
+```
+a=5
+b=6
+if [ $a -eq $b ]
+then echo "integer equality"
+elif [ $a -lt $b ]
+then echo "$a is smaller than $b"
+else echo "$a is greater than $b"
+fi
+```
+
+The statement `[ $a -lt $b ]` tests whether the variable `$a` is less than the variable `$b`. Similarly, `[ $a -gt $b ]` tests whether `$a` is greater than `$b`.
+
+<div class = "important">
+
+**Assigning Variables in Bash**
+
+Variables in Bash are assigned values using the equals symbol `=` with **no spaces** around it. When you want to refer to a variable later, use the dollar symbol `$` followed by the variable name, again with no spaces separating them.
+
+</div>
+
+You can have as many `elif` and `then` pairs as you want, but as soon as one is statement is true, the code will do that action stop. It will not continue to check the remaining statements. Try to think about what this code will output, then run it and check if you were right:
```

Comment:
```suggestion
You can have as many `elif` and `then` pairs as you want, but as soon as one is statement is true, the code will do that action **and then stop**. It will not continue to check the remaining statements! Try to think about what this code will output, then run it and check if you were right:
```